web/src/engine/osk/src/banner/bannerView.ts

```diff
@@ -178,6 +178,18 @@ export class BannerController {
     imagePath: ""
   }
 
+  // Default to black image banner for Android phone
+  public static readonly DEFAULT_ANDROID_PHONE_OPTIONS: BannerOptions = {
+    alwaysShow: true,
+    imagePath: "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABoAAAARCAIAAABM7ytaAAAAAXNSR0IArs4c6QAAAARnQU1BAACxjwv8YQUAAAAJcEhZcwAAEnQAABJ0Ad5mH3gAAAAVSURBVDhPYxgFo2AUjIJRQDpgYAAABT8AAcEGbxwAAAAASUVORK5CYII="
+  }
+
+  // Default to gray image banner for Android tablet
+  public static readonly DEFAULT_ANDROID_TABLET_OPTIONS: BannerOptions = {
+    alwaysShow: true,
+    imagePath: "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAB8AAAAXCAYAAADz/ZRUAAAABHNCSVQICAgIfAhkiAAAABl0RVh0U29mdHdhcmUAZ25vbWUtc2NyZWVuc2hvdO8Dvz4AAAAtdEVYdENyZWF0aW9uIFRpbWUARnJpIDA2IE9jdCAyMDIzIDAyOjUxOjQ0IFBNICswN4WPPMwAAABRSURBVEiJ7c1BCoAwDAXRaeP9b6PQs9kkbt02CEX4sx9eu86RCWQmK7k7c95ExNL37qAAA5gZZlaGAXoU4K/qbRsNfaMtXLhw4cKFCxf+Y/wBEz4WP2Yx4QAAAAAASUVORK5CYII=="
+  }
```

Comment:
I don't think we should be embedding PNGs as base64

Reply:
I was hoping all the images could be handled within KeymanWeb

Reply:
`imagePath` can take a static URL.

iOS uses the dataURL approach because it _dynamically builds_ the image at run time; the image exists first and foremost in memory.  `base64` becomes the best approach because of that, as we can then bypass saving the image to an actual file (and having to remember to delete it later).

Reply:
So I'm not currently using the .png files (yet) on this PR. I didn't know how to scale them for all the formfactors and orientations.

Even  in the Keyman for Android app, the "banner" at the top of the app is a composition of the logo with code generating the tri-color theme.
https://github.com/keymanapp/keyman/blob/e5ef4e9ee6fabab8f1b1b992bec12da25128e95d/android/KMAPro/kMAPro/src/main/java/com/tavultesoft/kmapro/MainActivity.java#L922-L955

Reply:
> So I'm not currently using the .png files (yet) on this PR. I didn't know how to scale them for all the formfactors and orientations.

I bet!  That's a chore even for iOS, which has a far more restricted range of resolution options.  Hence why iOS does the same for _both_ banners - both the app banner _and_ the keyboard banner.  (The latter of which is rendered to an image, then passed to Web.)

Reply:
Why don't we use .svg? We have a Keyman logo in SVG and that would mean just 1 image (perhaps 2, one for super-low res and one for everything else, but we have that too).

![keyman_logo_DETAIL (2)](https://github.com/keymanapp/keyman/assets/4498365/8c67b19c-7405-40f7-823c-40b0fc2a5b2d)
![keyman_logo_DETAIL](https://github.com/keymanapp/keyman/assets/4498365/9e1ac41f-b07a-4a0e-b60d-93db4ef08a48)

Both from Keyman Share Drive: Resources/Images/Keyman 9.0 Futago/Logo Pak (this folder could do with some tidyup, particularly we should make a well organised set of logo files in various formats!)

Reply:
Hmm.  That might work.  I'd have to double-check the banner's exact styling, but it shouldn't be hard to add a simple option to prevent stretching and/or tiling of the banner's image.

Reply:
While @mcdurdin and I were :duck: , he brought up the idea to see if @ermshiperete would be keen on updating KeymanWeb to render SVG's on the image banner. And keeping the Keyman logo within web/ :smile:

Reply:
Logo needs to be a Keyman for Android asset but should be handled entirely within the js/html with no Java interaction AFAICT. Remember  that other consumers will have their own logos for the banner.